Workaround for:

https://bugzilla.mozilla.org/show_bug.cgi?id=1450912

GCC libstdc++ should be fixed to support locales on Solaris so that
std::regex functionality works.

diff --git a/browser/app/nsBrowserApp.cpp b/browser/app/nsBrowserApp.cpp
index 6bab1ad..1abdf6d 100644
--- a/browser/app/nsBrowserApp.cpp
+++ b/browser/app/nsBrowserApp.cpp
@@ -246,6 +246,9 @@ uint32_t gBlocklistInitFlags = eDllBlocklistInitFlagDefault;
 #endif
 
 int main(int argc, char* argv[], char* envp[]) {
+
+  setenv("LC_ALL", "C", 1);
+
 #if defined(MOZ_ENABLE_FORKSERVER)
   if (strcmp(argv[argc - 1], "forkserver") == 0) {
     nsresult rv = InitXPCOMGlue(LibLoadingStrategy::NoReadAhead);
diff --git a/js/xpconnect/shell/xpcshell.cpp b/js/xpconnect/shell/xpcshell.cpp
index fe65bef..829a772 100644
--- a/js/xpconnect/shell/xpcshell.cpp
+++ b/js/xpconnect/shell/xpcshell.cpp
@@ -33,6 +33,9 @@
 #endif
 
 int main(int argc, char** argv, char** envp) {
+
+  setenv("LC_ALL", "C", 1);
+
 #ifdef MOZ_WIDGET_GTK
   // A default display may or may not be required for xpcshell tests, and so
   // is not created here. Instead we set the command line args, which is a
